json/src/de.rs

```diff
@@ -828,3 +829,53 @@ pub fn from_str<T>(s: &str) -> Result<T>
 {
     from_slice(s.as_bytes())
 }
+
+/// Iterator over JSON values
+pub struct JSONStream<T, Iter>
+    where Iter: Iterator<Item=io::Result<u8>>,
+          T: de::Deserialize
+{
+    deser: Deserializer<Iter>,
+    _marker: PhantomData<T>,
+}
+
+impl <T, Iter>JSONStream<T, Iter>
+    where Iter:Iterator<Item=io::Result<u8>>,
+          T: de::Deserialize {
+    fn new(i: Iter) -> JSONStream<T, Iter> {
+        JSONStream {
+            deser: Deserializer::new(i),
+            _marker: PhantomData
+        }
+    }
+}
+
+impl <T, Iter>Iterator for JSONStream<T, Iter>
+    where Iter:Iterator<Item=io::Result<u8>>,
+          T: de::Deserialize {
+    type Item = Result<T>;
+    fn next(&mut self) -> Option<Result<T>> {
+        match de::Deserialize::deserialize(&mut self.deser) {
+            Ok(v) => Some(Ok(v)),
+            Err(e) => {
+                match e {
+                    Error::SyntaxError(
+                        ErrorCode::EOFWhileParsingValue, _, _) =>
```

Comment:
I don't think this is the right thing do to here. It'll silently ignore a truncated json body. A more correct option would be to use a peek iterator, and before you parse the next json record, you can check if there's actually any bytes available to be read.